### PR TITLE
more use of emplace_back in write_resume_data

### PR DESCRIPTION
**Pull request by aldenml**

### CODE REVIEW

File: src/write_resume_data.cpp [hunk 1]
```diff
@@ -126,7 +126,7 @@ namespace libtorrent {
 		if (!atp.trackers.empty())
 		{
 			entry::list_type& tr_list = ret["trackers"].list();
-			tr_list.push_back(entry::list_type());
+			tr_list.emplace_back(entry::list_type());
```
Comment (arvidn) [inline src/write_resume_data.cpp]: when using ``emplace_back()``, I don't think you need to pass a default-initialized object. By not passing any arguments, it should just default initialize the ``list_type``, right?

Comment (aldenml) [inline src/write_resume_data.cpp]: yes, but below, it needs to be `entry::list_type`, can't be a generic default one (unless the default is a list, or...is it?)

File: src/write_resume_data.cpp [hunk 2]
```diff
@@ -232,7 +232,7 @@ namespace libtorrent {
 			// write file priorities
 			entry::list_type& prio = ret["file_priority"].list();
```
Comment (arvidn) [inline src/write_resume_data.cpp]: the more significant improvement here would probably be a ``prio.reserve()`` call

Comment (aldenml) [inline src/write_resume_data.cpp]: true

File: src/write_resume_data.cpp (hunk 1 above)
Comment (arvidn) [inline src/write_resume_data.cpp]: oh, right

**Comment by arvidn** [comment]
lgtm

**Pull request merged**